### PR TITLE
Redefine floatmin definition for BigFloat

### LINKED ISSUE #30387: inconsistent floatmin for `BigFloat`
This seems inconsistent with the definition of `floatmin` being the smallest non-subnormal value:
```jl
julia> [floatmin(T) == nextfloat(zero(T))/eps(T) for T in (Float16, Float32, Float64, BigFloat)]
4-element Array{Bool,1}:
  true
  true
  true
 false
```
Is this intentional?

Maybe we should define the fallback:
```jl
floatmin(::Type{T}) where {T<:AbstractFloat} = nextfloat(zero(T)) / eps(T)
```
cc @andreasnoack

### PR DESCRIPTION
**Pull request by sam0410**
Fixes #30387

### CODE REVIEW

File: base/float.jl [hunk 1]
```diff
@@ -727,14 +727,14 @@ end
...
-    floatmin(::Type{Float16}) = $(bitcast(Float16, 0x0400))
-    floatmin(::Type{Float32}) = $(bitcast(Float32, 0x00800000))
-    floatmin(::Type{Float64}) = $(bitcast(Float64, 0x0010000000000000))
+    
     floatmax(::Type{Float16}) = $(bitcast(Float16, 0x7bff))
     floatmax(::Type{Float32}) = $(bitcast(Float32, 0x7f7fffff))
     floatmax(::Type{Float64}) = $(bitcast(Float64, 0x7fefffffffffffff))
-
+    for T in (Float16, Float32, Float64)
+        @eval floatmin(T) = nextfloat(zero(T))/eps(T)
```
Comment (JeffBezanson) [inline base/float.jl]: This would need to be `::Type{$T}`, and `$T` on the other uses. But why not just keep the hard-coded values?

Comment (sam0410) [reply base/float.jl]: I suppose I misunderstood [this comment](https://github.com/JuliaLang/julia/issues/30387#issuecomment-447958221). I made the change only in Bigfloat. Is this what is expected?

Sorry for the trouble.

**Comment by ViralBShah** [comment]
Definitely needs a better commit message before merging.

**Comment by sam0410** [comment]
Didn't see a pull was already created in  #30393

**Pull request closed without merging**